### PR TITLE
Remove duplicate hashing of passwords

### PR DESCRIPTION
**Pull request by AlexanderRMcNeill**
## What
This PR removes the duplicate hashing of passwords. 

The password attributes for the `resource_artifactory_remote_repository` and `resource_artifactory_replication_config` resources both have a `StateFunc`. This function is used to change the value of the given attribute before it is saved to state (and likewise before comparing for diffs). The code also hashes the passwords before setting them in state. This means that the hashed password in state has been hashed twice and will always be different than the config value which has only been hashed once.

## Why
With the current implementation there will always be a difference even if the password hasn't changed.

### CODE REVIEW

**Comment by AlexanderRMcNeill** [comment]
My initial change didn't fully fix the issue. It stop the duplicate hashing of the passwords stored in state but there is still an issue causing the password to always be different. The issue is the password returned from the API is encrypted. 

To get around this issue I've updated the code to use the configuration as the source of truth instead of the password returned from the API. This means that update function will only be called when there is a difference between the password that is stored in state and the one that is in the config.

**Comment by AlexanderRMcNeill** [comment]
Also, I'm unable to run the acceptance tests sure to the licence in the repo not working.

File: pkg/artifactory/resource_artifactory_single_replication_config.go [hunk 1]
```diff
@@ -3,10 +3,12 @@ package artifactory
 import (
 	"context"
 	"fmt"
+	"net/http"
+
 	"github.com/atlassian/go-artifactory/v2/artifactory"
 	"github.com/atlassian/go-artifactory/v2/artifactory/v1"
+	v1 "github.com/atlassian/go-artifactory/v2/artifactory/v1"
```
Comment (jamestoyer) [inline pkg/artifactory/resource_artifactory_single_replication_config.go:9]: It looks like your tests in Travis are failing due to the inclusion of this line. It's the same as the previous line. If you try removing it, the test should run again

For reference the error in Travis was
```
pkg/artifactory/resource_artifactory_single_replication_config.go:10:2: v1 redeclared as imported package name
	previous declaration at pkg/artifactory/resource_artifactory_single_replication_config.go:9:2
FAIL	github.com/atlassian/terraform-provider-artifactory/pkg/artifactory [build failed]
```

Comment (AlexanderRMcNeill) [reply pkg/artifactory/resource_artifactory_single_replication_config.go:9]: I've removed the duplicate import. My editor really wants to be "helpful" and add the alias for v1 even though it doesn't really matter. I can remove it if you want.

**Review by dillon-giacoppo — APPROVED** [review]
Thanks for contributing. Whilst the double hashing is a bug, I don't think this is going to fix the behaviour of passwords being updated every time.

The reason you are probably seeing passwords requiring updates is because of https://github.com/atlassian/terraform-provider-artifactory/issues/24. There are a couple workarounds there whilst support for encrypted passwords is developed.

**Comment by AlexanderRMcNeill** [comment]
@dillon-giacoppo yeah, after making the change and testing it I realised it didn't fix the issue. In my second commit I updated it to set the password to be from terraform configuration instead of the password returned from the API.

```
		if d.Get("password") != nil {
			logErr(d.Set("password", d.Get("password").(string)))
		} else {
			logErr(d.Set("password", *repo.Password))
		}
```

This means that it will only update if there is a difference between the configuration and the state file.

Sorry about making this confusing by doing essentially two different things in this PR. I'm happy to update this to just be the double hashing fix and make another PR to fix the password diff issue.

**Comment by AlexanderRMcNeill** [comment]
Closed due to PR doing multiple things. I will open separate PRs for fixing the password hashing and updating the diff to be with the state file instead of the value returned from the API

**Pull request closed without merging**